controllers/ocirepository_controller.go

```diff
@@ -347,12 +344,11 @@ func (r *OCIRepositoryReconciler) reconcileSource(ctx context.Context, obj *sour
 		conditions.MarkTrue(obj, sourcev1.FetchFailedCondition, e.Reason, e.Err.Error())
 		return sreconcile.ResultEmpty, e
 	}
-	if transport != nil {
-		options = append(options, crane.WithTransport(transport))
-	}
+
+	opts := r.makeOptions(ctx, obj, withTransport(transport), withKeychainOrAuth(keychain, auth))
```

Comment:
Wouldn't it be better to simply make the signature here `makeOptions(ctx, obj, transport, keychain, auth)` to prevent another options interface, and just have flat out conditional logic in `makeOptions`?

Reply:
We will likely keep adding options (at least for `insecure`).

Reply:
I have added a commit that complies with this.